exercises/currying/Curry1.hs

```diff
@@ -0,0 +1,49 @@
+module Curry1 where
+
+-- I AM NOT DONE
+
+{-
+- In Haskell, functions that take more than one argument are usually
+  curried, meaning the function takes one argument and returns another function that takes the second argument. 
+  
+- The standard library defines a *higher order function* *curry* which 
+  takes a function from a pair (a,b) -> c as an argument and returns a function that takes a and returns a function b -> c.
```

Comment:
This concept is tricky enough that we should probably use 2-3 sentences :) Maybe something like:

1. The standard library defines a higher order function curry.
2. This functions takes a single input argument: a function that takes a pair `(a, b)` as input.
3. The result of this function is a *new function* that takes its inputs separately (`a -> b -> c`), rather than as a tuple/pair.

Reply:
I'm happy with your suggestions. I agree it would fit well with functions, as it's kind of fundamental to Haskell functions that they are usually curried. This will require getting rid of the type synonym.

If we put it in the functions directory, we have to make sure to change it in the ExerciseList. It's currently saying 'currying'. 

Are you making the changes or would you like me to do it? I don't mind either way.

Reply:
Oh you can do the changes!